src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Bindings.Maven.targets

```diff
@@ -0,0 +1,35 @@
+<!--
+***********************************************************************************************
+Xamarin.Android.Bindings.Maven.targets
+
+This file contains MSBuild targets used to enable @(AndroidMavenLibrary) support.
+
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="Xamarin.Android.Tasks.MavenDownloadTask" AssemblyFile="Xamarin.Android.Build.Tasks.dll" />
+
+  <PropertyGroup>
+    <!-- Maven cache directory, locations copied from XBD which should be sufficiently battle-tested -->
+    <MavenCacheDirectory Condition="'$(OS)'=='Unix' and '$(MavenCacheDirectory)'==''">$(HOME)\Library\Caches\MavenCacheDirectory\</MavenCacheDirectory>
+    <MavenCacheDirectory Condition="'$(OS)'!='Unix' and '$(MavenCacheDirectory)'==''">$(LocalAppData)\MavenCacheDirectory\</MavenCacheDirectory>
+    <MavenCacheDirectory Condition="'$(MavenCacheDirectory)' != '' and !HasTrailingSlash('$(MavenCacheDirectory)')">$(MavenCacheDirectory)\</MavenCacheDirectory>
+  </PropertyGroup>
+
+  <Target Name="MavenRestore" BeforeTargets="_CategorizeAndroidLibraries" DependsOnTargets="ResolvePackageAssets">
```

Comment:
This also runs the `MavenRestore` target on *every* build. Is there a way it can be incremental with `Inputs/Outputs`, so it only runs if some file changes?

See: https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/MSBuildBestPractices.md#incremental-builds

Reply:
I do not think it will be possible to make this incremental, as we do not have a way to determine what the outputs will be.  For example, `androidx.core:core` could resolve to `androidx-core.jar` or `androidx-core.aar`.

I would suspect that running this when everything will be found in the cache should be trivial, but testing that theory is a good idea.

Reply:
I think you can write a stamp file as the above doc mentions. It keeps a timestamp of the last time the target ran, and could only rerun if a `@(AndroidMavenLibrary)` changes -- or some other set of files change.

Reply:
We still need the output of the task in order to run.  That is, given `androidx.core:core`, we need to resolve it to the `.jar`/`.aar` that exists in the cache so we can pass it to `@(AndroidLibrary)`.

My hunch is we are talking about ~1ms, but if it turns out to be a significant amount of time we can definitely revisit.

Reply:
No, you don't necessarily need to make the .jar/aar file the output, it can be a stamp file:

https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/MSBuildBestPractices.md#stamp-files

This is just a marker to know the last time the target ran.

Reply:
Right, but I think we *need* the Output parameter (`@(ResolvedAndroidMavenLibraries)`) from `MavenDownloadTask` in order to pass to `@(AndroidLibrary)`.

That is, something has to convert:

```xml
<AndroidMavenLibrary Include="com.google.auto.value:auto-value-annotations" Version="1.10.4" />
```

to:

```xml
<AndroidLibrary Include="$(LocalAppData)\MavenCacheDirectory\Central\com.google.auto.value\auto-value-annotations\auto-value-annotations.jar" />
```

If this doesn't run, `@(AndroidLibrary.Count())` will be `0` and I assume the rest of the binding build will fall apart?

For example, if the user just changed a metadata file, this target would need to be run or else the build would not have a `.jar`/`.aar` to try to bind.

Reply:
You could save the `[Output]` in a file to use in incremental builds. We do something like this here, but this code has been here a long time:

https://github.com/xamarin/xamarin-android/blob/07dc60b04273941787e56ff3d02fbb8df98ab0ce/src/Xamarin.Android.Build.Tasks/Xamarin.Android.EmbeddedResource.targets#L60-L67

Maybe making this incremental could be a "part 2" PR; we might could merge this one and find out how slow it is first.

Reply:
I ran some quick tests.  The cached run is slower than anticipated, I think this is because it loads `MavenNet.dll` from disk.  Running many cases points to a fixed overhead.

| Scenario | Uncached | Cached
|-|-|-|
| 1 MavenAndroidLibrary | 77ms | 8ms |
| 29 MavenAndroidLibraries | 5372ms | 13ms

I think this is probably still in the acceptable range.